### ISSUE TITLE
Is Windows supported?

### ISSUE THREAD

**Issue by MatanRubin** [petereon/beaupy#30]
Hi,
Awesome project - really love how simple the APIs are and how quickly I could create a fairly complex app. Thank you for your work!

I couldn't see any reference to it in the docs, so I thought I'd just ask here - is Windows supported? I'm developing on a Mac but my app targets Windows users as well - can I expect this to work equally well on Windows environments?

Thanks!

**Comment by petereon** [comment]
Hi,
thank you for the kind words, it is much appreciated!

As for the question of OS support, great question, I've never really had a proper look, I'd do well to add that into the documentation. 

Anyways, `beaupy` is tested against all the major platforms but all of the console interaction is mocked. It relies on the dependencies to handle the interaction with the console, namely [readchar](https://github.com/magmax/python-readchar) to handle user input and [rich](https://github.com/Textualize/rich) for console printing. As far as I've checked `readchar` officially tests against Linux and Windows but noting all Unix based systems should probably work[^1] (I've just tested against MacOS and had no issues) and `rich` officially supports all major platforms[^2]. 

To summarize, Windows and Linux are officially supported by each and all of the other Unix-based operating systems probably work.

That said, I'd still recommend to test against the platform your target audience is going to use.

[^1]: https://github.com/magmax/python-readchar#os-support
[^2]: https://github.com/Textualize/rich#compatibility

**Comment by petereon** [comment]
@MatanRubin, has your question been answered? Can I consider this closed?

**Comment by MatanRubin** [comment]
Feel free to close it for now. I'll let some beta testers play with my app on Windows soon and will reopen this issue if we find any issues.

Thanks!

**Comment by petereon** [comment]
Thanks! If you experience any issues, free to open a new issue.
